tests/longevity/longevity_test.go

```diff
@@ -0,0 +1,342 @@
+package tests
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/reporters"
+	. "github.com/onsi/gomega"
+	"github.com/portworx/sched-ops/k8s/core"
+	"github.com/portworx/torpedo/drivers/scheduler"
+	. "github.com/portworx/torpedo/tests"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	testTriggersConfigMap = "longevity-triggers"
+	configMapNS           = "default"
+)
+
+var (
+	// Stores mapping between test trigger and its chaos level.
+	chaosMap map[string]int
+	// Stores mapping between chaos level and its freq. Values are hardcoded
+	triggerInterval map[string]map[int]time.Duration
+	// Stores which are disruptive triggers. When disruptive triggers are happening in test,
+	// other triggers are allowed to happen only after existing triggers are completed.
+	disruptiveTriggers map[string]bool
+)
+
+func TestLongevity(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	var specReporters []Reporter
+	junitReporter := reporters.NewJUnitReporter("/testresults/junit_basic.xml")
+	specReporters = append(specReporters, junitReporter)
+	RunSpecsWithDefaultAndCustomReporters(t, "Torpedo : Longevity", specReporters)
+}
+
+var _ = BeforeSuite(func() {
+	InitInstance()
+	populateIntervals()
+	populateDisruptiveTriggers()
+})
+
+var _ = Describe("{Longevity}", func() {
+	var contexts []*scheduler.Context
+	var triggerLock sync.Mutex
+	triggerEventsChan := make(chan *EventRecord, 100)
+	triggerFunctions := map[string]func([]*scheduler.Context, *chan *EventRecord){
+		RebootNode:       TriggerRebootNodes,
+		RestartVolDriver: TriggerRestartVolDriver,
+		CrashVolDriver:   TriggerCrashVolDriver,
+		HAUpdate:         TriggerHAUpdate,
+		EmailReporter:    TriggerEmailReporter,
+		AppTaskDown:      TriggerAppTaskDown,
+	}
+	It("has to schedule app and introduce test triggers", func() {
+		Step(fmt.Sprintf("Start watch on K8S configMap [%s] in namespace [%s]",
```

Comment:
nit: change to `... K8S configMap [%s/%s]", configMapNS , testTriggersConfigMap `